docking.js

```diff
@@ -655,23 +654,26 @@ var DockedDash = GObject.registerClass({
     }
 
     _hoverChanged() {
-        if (!this._ignoreHover) {
-            // Skip if dock is not in autohide mode for instance because it is shown
-            // by intellihide.
-            if (this._autohideIsEnabled) {
-                if (this._box.hover)
-                    this._show();
-                else
-                    this._hide();
-            }
+        // Skip if dock is not in autohide mode for instance because it is shown by intellihide or
+        // if the dock's `_ignoreHover` value is set to true.
+        if (this._ignoreHover || !this._autohideIsEnabled) {
+            return;
         }
+
+        if (this._triggerTimeoutId)
+            this._isPointerInZone() || this._box.hover ? this._show() : this._hide();
+        else
+            this._box.hover ? this._show() : this._hide();
     }
 
     getDockState() {
         return this._dockState;
     }
 
     _show() {
+        // Remove any delayed hide animation.
+        delete this._delayedHide;
+
```

Comment:
Delete the `_delayedHide` value in `_show` instead of `_animateIn` since calls to said `_show` are ignored when the dock's state is either "shown" or "showing" and, when a delayed hide transition is queued, the dock's state is only updated AFTER the animate-in transition is completed.